instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/patches/client.rb

```diff
@@ -9,47 +9,8 @@ module Instrumentation
     module Mysql2
       module Patches
         # Module to prepend to Mysql2::Client for instrumentation
-        module Client # rubocop:disable Metrics/ModuleLength
-          QUERY_NAMES = [
-            'set names',
-            'select',
-            'insert',
-            'update',
-            'delete',
-            'begin',
-            'commit',
-            'rollback',
-            'savepoint',
-            'release savepoint',
-            'explain',
-            'drop database',
-            'drop table',
-            'create database',
-            'create table'
-          ].freeze
-
-          QUERY_NAME_RE = Regexp.new("^(#{QUERY_NAMES.join('|')})", Regexp::IGNORECASE)
-
-          # From: https://github.com/newrelic/newrelic-ruby-agent/blob/0235b288d85b8bc795bdc1a24621dd9f84cfef45/lib/new_relic/agent/database/obfuscation_helpers.rb#L9-L34
-          COMPONENTS_REGEX_MAP = {
-            single_quotes: /'(?:[^']|'')*?(?:\\'.*|'(?!'))/,
-            double_quotes: /"(?:[^"]|"")*?(?:\\".*|"(?!"))/,
-            numeric_literals: /-?\b(?:[0-9]+\.)?[0-9]+([eE][+-]?[0-9]+)?\b/,
-            boolean_literals: /\b(?:true|false|null)\b/i,
-            hexadecimal_literals: /0x[0-9a-fA-F]+/,
-            comments: /(?:#|--).*?(?=\r|\n|$)/i,
-            multi_line_comments: %r{\/\*(?:[^\/]|\/[^*])*?(?:\*\/|\/\*.*)}
-          }.freeze
-
-          MYSQL_COMPONENTS = %i[
-            single_quotes
-            double_quotes
-            numeric_literals
-            boolean_literals
-            hexadecimal_literals
-            comments
-            multi_line_comments
-          ].freeze
+        module Client
+          include OpenTelemetry::Instrumentation::Helpers::MysqlHelper
```

Comment:
I'm reluctant to use includes on classes that are patching other classes. We have a helper including a helper which in turn gets included here. My concern is the easy of potentially having any of those modules have method names that collide with the class being patched, maybe not initially with this PR, but as time progresses. 

Instead I'd be more inclined for us to require the helper and explicitly reference the methods like `OpenTelemetry::Instrumentation::Helpers::MysqlHelper. generated_regex`, it is very wordy, but it's perfectly clear where the code is coming from.

Reply:
There's still some work to do to get this PR ready to go, but the shift to classes and independent helper gems has been made!